include/seqan3/alignment/pairwise/execution/alignment_executor_two_way.hpp

```diff
@@ -257,27 +257,25 @@ class alignment_executor_two_way
         // Apply the alignment execution.
         size_t count = 0;
         size_t buffer_limit = in_avail();
-        assert(buffer_limit >= chunk_size());
 
-        while (count < buffer_limit && !is_eof())
+        size_t current_chunk_size = std::ranges::distance(*chunked_resource_it);
```

Comment:
maybe assert that the `chunked_resource_it` is a forward_iterator? might save some debugging time if we ever use a wrong view on top

Reply:
and you can set this variable inside of the for loop signature.

Reply:
> maybe assert that the chunked_resource_it is a forward_iterator? might save some debugging time if we ever use a wrong view on top

I don't see this as a necessary step as the underlying range is required to be forward range. I don't see why chunk view should ever degrade a forward range to an input range.